packages/runner/src/traverse.ts

```diff
@@ -625,6 +626,52 @@ export class SchemaObjectTraverser<K, S> extends BaseObjectTraverser<K, S> {
     }
   }
 
+  static resolveSchemaRef(
+    schemaContext: Readonly<SchemaContext>,
+  ): Readonly<SchemaContext> | undefined {
+    if (
+      !isObject(schemaContext.schema) || !("$ref" in schemaContext.schema) ||
+      schemaContext.schema["$ref"] === undefined
+    ) {
+      // We shouldn't have called this function, but just return it as-is
+      return schemaContext;
+    }
+    if (schemaContext.rootSchema === undefined) {
+      logger.warn(() => ["Unsupported $ref without root schema: ", schemaRef]);
+      return undefined;
+    }
+    const schemaRef = schemaContext.schema["$ref"];
+    if (schemaRef === "#") {
+      return {
+        schema: schemaContext.rootSchema,
+        rootSchema: schemaContext.rootSchema,
+      };
+    } else if (
+      schemaRef.startsWith("#/definitions/") || schemaRef.startsWith("#/$defs/")
+    ) {
+      const pathToDef = schemaRef.split("/");
+      let schemaCursor = schemaContext.rootSchema;
+      // start at 1, since the 0 element is "#"
+      for (let i = 1; i < pathToDef.length; i++) {
+        if (!isObject(schemaCursor) || !(pathToDef[i] in schemaCursor)) {
```

Comment:
super nit, but are arrays explicitly not allowed in the standard? use `isRecord` and then it's ok to be either an array or an object